modules/cluster-management/pom.xml

```diff
@@ -49,17 +49,26 @@
 
         <dependency>
             <groupId>org.apache.ignite</groupId>
-            <artifactId>ignite-rest</artifactId>
+            <artifactId>ignite-vault</artifactId>
         </dependency>
 
         <dependency>
             <groupId>org.apache.ignite</groupId>
-            <artifactId>ignite-vault</artifactId>
+            <artifactId>ignite-rest-api</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.micronaut</groupId>
+            <artifactId>micronaut-http</artifactId>
+            <version>${micronaut.version}</version>
```

Comment:
As for micronaut-bom, it might not pass vulnerabilities checks https://mvnrepository.com/artifact/io.micronaut/micronaut-bom/3.5.0. 

One more thing is that it declares a huge set of dependencies that are not only micronaut-related. That could be a problem in dependency management.
https://repo1.maven.org/maven2/io/micronaut/micronaut-bom/3.5.0/micronaut-bom-3.5.0.pom

Reply:
> As for micronaut-bom, it might not pass vulnerabilities checks

Why? There should be no difference between using a bom and declaring dependencies directly

> That could be a problem in dependency management.

Can you provide an example please?

Reply:
> Why? There should be no difference between using a bom and declaring dependencies directly

Yes, you are right, there is no difference.

> Can you provide an example please?

Micronaut-bom includes a lot of other libraries' version definitions, for example, neo4j.version, h2.version, and many more. So, Ignite 3 will have them included to dependency management too.  Maybe that is not a problem but if Iginite 3 would like to support neo4j with a version that is different from micronaut-bom then we might face an issue.

Reply:
AFAIK, if you declare a dependency directly in the dependency management section, it will override any artifacts declared in imported boms

Reply:
What about two boms have different versions?